File: src/main/java/com/github/weewar/mapviewer/service/ImageRepository.java
package com.github.weewar.mapviewer.service;

import com.github.weewar.mapviewer.exceptions.ImageNotFoundException;
import com.github.weewar.mapviewer.exceptions.ImagePreloadException;
import com.github.weewar.mapviewer.model.enums.Direction;
import com.github.weewar.mapviewer.model.enums.Owner;
import com.github.weewar.mapviewer.model.enums.TerrainType;
import com.github.weewar.mapviewer.model.enums.UnitType;

import java.awt.*;

public interface ImageRepository {
    Image getTerrain(TerrainType terrainType, Owner owner, Direction direction) throws ImageNotFoundException;

    Image getUnit(UnitType terrainType, Owner owner) throws ImageNotFoundException;

    void preloadImages() throws ImagePreloadException;
}

File: src/main/kotlin/com/github/weewar/mapviewer/service/ImageRepository.kt
package com.github.weewar.mapviewer.service

import com.github.weewar.mapviewer.model.ImageNotFoundException
import com.github.weewar.mapviewer.model.ImagePreloadException
import com.github.weewar.mapviewer.model.enums.*
import java.awt.Image

interface ImageRepository {
    @Throws(ImageNotFoundException::class)
    fun getTerrain(terrainType: TerrainType, owner: Owner?, direction: Direction?): Image

    @Throws(ImageNotFoundException::class)
    fun getUnit(unitType: UnitType, owner: Owner): Image

    @Throws(ImagePreloadException::class)
    fun preloadImages()
}

